google/cloud/bigquery/dtypes.py

```diff
@@ -0,0 +1,471 @@
+        keepdims: bool = False,
+        skipna: bool = True,
+    ):
+        pandas.compat.numpy.function.validate_any(
+            (), {"out": out, "keepdims": keepdims}
+        )
+        result = pandas.core.nanops.nanany(self._ndarray, axis=axis, skipna=skipna)
+        return result
+
+    def all(
+        self,
+        *,
+        axis: Optional[int] = None,
+        out=None,
+        keepdims: bool = False,
+        skipna: bool = True,
+    ):
+        pandas.compat.numpy.function.validate_all(
+            (), {"out": out, "keepdims": keepdims}
+        )
+        result = pandas.core.nanops.nanall(self._ndarray, axis=axis, skipna=skipna)
+        return result
+
+    def min(
+        self, *, axis: Optional[int] = None, skipna: bool = True, **kwargs
+    ) -> Scalar:
+        pandas.compat.numpy.function.validate_min((), kwargs)
+        result = pandas.core.nanops.nanmin(
+            values=self._ndarray, axis=axis, mask=self.isna(), skipna=skipna
+        )
+        return self._box_func(result)
+
+    def max(
+        self, *, axis: Optional[int] = None, skipna: bool = True, **kwargs
+    ) -> Scalar:
+        pandas.compat.numpy.function.validate_max((), kwargs)
+        result = pandas.core.nanops.nanmax(
+            values=self._ndarray, axis=axis, mask=self.isna(), skipna=skipna
+        )
+        return self._box_func(result)
+
+    if pandas_release >= (1, 2):
+
+        def median(
+            self,
+            *,
+            axis: Optional[int] = None,
+            out=None,
+            overwrite_input: bool = False,
+            keepdims: bool = False,
+            skipna: bool = True,
+        ):
+            pandas.compat.numpy.function.validate_median(
+                (),
+                {"out": out, "overwrite_input": overwrite_input, "keepdims": keepdims},
+            )
+            result = pandas.core.nanops.nanmedian(
+                self._ndarray, axis=axis, skipna=skipna
+            )
+            return self._box_func(result)
+
+
+@pandas.core.dtypes.dtypes.register_extension_dtype
+class TimeDtype(_BaseDtype):
+    """
+    Extension dtype for time data.
+    """
+
+    name = time_dtype_name
+    type = datetime.time
+
+    def construct_array_type(self):
+        return TimeArray
+
+
+class TimeArray(_BaseArray):
+    """
+    Pandas array type containing time data
+    """
+
+    # Data are stored as datetime64 values with a date of Jan 1, 1970
+
+    dtype = TimeDtype()
+    _epoch = datetime.datetime(1970, 1, 1)
+    _npepoch = numpy.datetime64(_epoch)
+
+    @classmethod
+    def _datetime(cls, scalar):
+        if isinstance(scalar, datetime.time):
+            return datetime.datetime.combine(cls._epoch, scalar)
+        elif isinstance(scalar, str):
+            # iso string
+            h, m, s = map(float, scalar.split(":"))
+            s, us = divmod(s, 1)
+            return datetime.datetime(
+                1970, 1, 1, int(h), int(m), int(s), int(us * 1000000)
```

Comment:
(nit) There are a bit too many zeros to parse this easily, let's add separators.
```suggestion
                1970, 1, 1, int(h), int(m), int(s), int(us * 1_000_000)
```

Reply:
https://github.com/googleapis/python-db-dtypes-pandas/pull/9/commits/88a87fc8440702bdc8ced7a7e4bd93a02c8fa1fc